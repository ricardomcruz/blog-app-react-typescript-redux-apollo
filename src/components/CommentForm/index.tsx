import classNamesBind from 'classnames/bind';
import { useState } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { RawComment } from '../../_modules/comments/models/Comment';
import styles from './styles.module.scss';

const cx = classNamesBind.bind(styles);
interface CommentFormProps {
  postId: number,
  onCommentSubmit: (data: RawComment) => void
}

const CommentForm = (
  {
    postId,
    onCommentSubmit
  }: CommentFormProps
) => {


  const date = new Date().toISOString().slice(0, 10);;

  const emptyComment = {
    postId,
    user: '',
    date,
    content: ''
  }

  const [value, setValue] = useState<RawComment>(emptyComment as RawComment);
  const [validated, setValidated] = useState(false);
  const [sent, setSent] = useState(false);

  const handleChange = (event: any, field: string) => {
    setValue({ ...value, [field]: event.target.value});
    setValidated(false);
    setSent(false);
  }

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    const form = event.currentTarget;

    
    if (form.checkValidity() === false) {
      event.preventDefault();
      event.stopPropagation();
    }
    else {
      setSent(true);
      onCommentSubmit(value)
    }
    setValidated(true);
  };

  return (
    <Form noValidate validated={validated} onSubmit={handleSubmit}>
      <Form.Group
        className="mb-3"
        controlId="validationCustom01"
        onChange={(e) => handleChange(e, 'user')}
      >
        <Form.Label>Name</Form.Label>
        <Form.Control required type="text" />
        <Form.Control.Feedback type="invalid">
          Please provide your name
        </Form.Control.Feedback>
      </Form.Group>
      <Form.Group
        onChange={(e) => handleChange(e, 'content')}
        className="mb-3"
        controlId="validationCustom02"
      >
        <Form.Label>Comment</Form.Label>
        <Form.Control required as="textarea" rows={3} />
        <Form.Control.Feedback type="invalid">
          Please provide your comment
        </Form.Control.Feedback>
      </Form.Group>
      <div className={cx('button-container')}>
        <Button type="submit">{
          sent ?
            'Sent'
            :
            'Submit form'
        }
        </Button>
      </div>
      
    </Form>
  );
}

export default CommentForm;
