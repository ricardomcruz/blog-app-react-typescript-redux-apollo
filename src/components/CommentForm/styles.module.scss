// Required
@import "../../globals.scss";


.post-card {
  height: 100%;

  &-title {
    text-decoration: inherit;
  }

  &:hover {
    border: 1px solid blue;
  }
}