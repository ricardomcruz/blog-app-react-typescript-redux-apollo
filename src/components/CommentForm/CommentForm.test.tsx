/** * @jest-environment jsdom */

import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import CommentForm from './index';

describe('CommentForm', () => {

  const handleCommentSubmit = jest.fn();

  it('should render the Comment Form', () => {
    const { asFragment }= render(
      <CommentForm
        postId={1}
        onCommentSubmit={handleCommentSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
