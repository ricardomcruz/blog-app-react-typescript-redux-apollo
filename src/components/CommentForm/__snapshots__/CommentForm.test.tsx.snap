// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm should render the Comment Form 1`] = `
<DocumentFragment>
  <form
    class=""
    novalidate=""
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="validationCustom01"
      >
        Name
      </label>
      <input
        class="form-control"
        id="validationCustom01"
        required=""
        type="text"
      />
      <div
        class="invalid-feedback"
      >
        Please provide your name
      </div>
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="validationCustom02"
      >
        Comment
      </label>
      <textarea
        class="form-control"
        id="validationCustom02"
        required=""
        rows="3"
      />
      <div
        class="invalid-feedback"
      >
        Please provide your comment
      </div>
    </div>
    <div
      class="button-container"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit form
      </button>
    </div>
  </form>
</DocumentFragment>
`;
